fenix/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt

```diff
@@ -363,19 +362,8 @@ object TestHelper {
 
     // Permission deny dialogs differ on various Android APIs
     fun denyPermission() {
-        if (Build.VERSION.SDK_INT >= 23) {
-            mDevice.findObject(
-                By.text(
-                    when (Build.VERSION.SDK_INT) {
-                        Build.VERSION_CODES.R -> Pattern.compile(
-                            "DENY",
-                            Pattern.CASE_INSENSITIVE,
-                        )
-                        else -> Pattern.compile("Deny", Pattern.CASE_INSENSITIVE)
-                    },
-                ),
-            ).click()
-        }
+        mDevice.findObject(UiSelector().textContains("Deny")).waitForExists(waitingTime)
+        mDevice.findObject(UiSelector().textContains("Deny")).click()
```

Comment:
Didn't we need the case sensitive pattern matching because we saw the case changes on different Android environments? 

Maybe can just swap to `textStartsWith` as it's case insensitive? 
https://developer.android.com/reference/androidx/test/uiautomator/UiSelector#textStartsWith(java.lang.String)

Reply:
https://issuetracker.google.com/issues/133770809

Seems to be that `textContains` is fine, and the doc is wrong 🤷🏼